src/test/scala/uk/gov/hmrc/test/ui/pages/section6/InlandModeOfTransportPage.scala

```diff
@@ -19,12 +19,18 @@ package uk.gov.hmrc.test.ui.pages.section6
 import uk.gov.hmrc.test.ui.pages.base.Constants.Common
 import uk.gov.hmrc.test.ui.pages.base.TariffLinks.inlandTransportDetails
 import uk.gov.hmrc.test.ui.pages.base.{BasePage, Detail}
-import uk.gov.hmrc.test.ui.pages.section6.DetailKeys.{InlandModeOfTransport, InlandOrBorder}
+import uk.gov.hmrc.test.ui.pages.section6.DetailKeys.{InlandModeOfTransport, InlandOrBorder, TransportLeavingBorder}
 
 object InlandModeOfTransportPage extends BasePage {
 
-  def backButtonHref: String = maybeDetail(InlandOrBorder).fold(InlandOrBorderPage.backButtonHref)(_ => InlandOrBorderPage.path)
-  val path: String  = "/declaration/inland-transport-details"
+  def backButtonHref: String =
+    if (detail(TransportLeavingBorder).equals("Roll on Roll off (RoRo)"))
```

Comment:
This is Scala not Java :smiley_cat: please `if (detail(TransportLeavingBorder) == "Roll on Roll off (RoRo)")`

Still, I would rather place the logic in a specific method. Something like...
`def isRoroTransportLeavingTheBorder: Boolean = detail(TransportLeavingBorder) == "Roll on Roll off (RoRo)"`